### PR TITLE
Fix logger initialization

### PR DESCRIPTION
**Pull request by matbun**
# Summary

- improve the mechanism used to ensure that a logger is used after having been properly initialized
- make the logger wrapper/adapter for Lightning more robust by avoiding the corner case in which a logger is initialized twice
- add tests on initialization

---

**Related issue :** #283

### CODE REVIEW

File: src/itwinai/loggers.py [hunk 1]
```diff
@@ -672,18 +723,21 @@ def __init__(
...
-                used in distributed environments. Defaults to None.
+            rank (int): global rank of current process,
+                used in distributed environments. Defaults to 0.
         """
         self.worker_rank = rank
 
         if not self.should_log():
             return
 
+        logging.info(f"Initializing {self.__class__.__name__} on rank {rank}")
```
Comment (annaelisalappe) [inline src/itwinai/loggers.py:739]: Have you thought about putting
```
 self.worker_rank = rank

        if not self.should_log():
            return

        logging.info(f"Initializing {self.__class__.__name__} on rank {rank}")
```
into the parent function? Then you can do `super(). create_logger_context(rank)` and you don't have to call it in every logger again.

File: tests/loggers/test_loggers.py [hunk 2]
```diff
@@ -7,12 +7,51 @@
...
+    assert "has not been initialized" in str(exc_info.value)
+    with pytest.raises(RuntimeError) as exc_info:
+        itwinai_logger.destroy_logger_context()
+    assert "has not been initialized" in str(exc_info.value)
+
+    itwinai_logger.create_logger_context()
+
+    # Double initialization
+    caplog.clear()
+    with caplog.at_level(logging.WARNING):
```
Comment (annaelisalappe) [inline tests/loggers/test_loggers.py:45]: This is cool :)

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -55,6 +55,9 @@ jobs:
           VALIDATE_JSON_PRETTIER: false
           VALIDATE_CSS_PRETTIER: false
           VALIDATE_KUBERNETES_KUBECONFORM: false
+          # Both options below should be already covered by ruff
+          VALIDATE_PYTHON_ISORT: false
+          VALIDATE_PYTHON_FLAKE8: false
```
Comment (annaelisalappe) [inline .github/workflows/lint.yml:60]: Is this from an old commit or did you change this?

Comment (matbun) [reply .github/workflows/lint.yml:60]: I had to change it because isort + flake8 were conflicting with ruff on how to better lint some imports... idk if you ever noticed sth similar

File: src/itwinai/loggers.py (hunk 1 above)
Comment (matbun) [reply src/itwinai/loggers.py:744]: yes, good idea

Comment (matbun) [reply src/itwinai/loggers.py:744]: I wouldn't know how to implement the "return" logic to skip logging when not supposed to.. If I return in super() I just go back to the caller

File: .github/workflows/lint.yml (hunk 3 above)
Comment (annaelisalappe) [reply .github/workflows/lint.yml:60]: Ah okay.. I had, but I changed these lines ages ago I though, that's why I asked if it was old - weird

Comment (matbun) [reply .github/workflows/lint.yml:60]: ok. well, hopefully it'll stay this time :)

File: src/itwinai/loggers.py (hunk 1 above)
Comment (annaelisalappe) [reply src/itwinai/loggers.py:744]: Ah, I see. Well I suppose it's fine to leave it as it is as well 😌
